.github/workflows/ci.yml

```
@@ -0,0 +1,56 @@
+name: CI
+
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  push:
+    branches:
+      - 'main'
+    tags: ["v[0-9]+.[0-9]+[0-9]+"]
+  pull_request:
+    branches:
+      - 'main'
+
```

I recommend adding `workflow_dispatch` so you can start the workflow manually on GitHub.

```suggestion

  workflow_dispatch:
  
```